Src/VimCore/CoreInterfaces.fs

```diff
@@ -1509,61 +1509,82 @@ type KeyMappingResult =
 [<DebuggerDisplay("{ToString()}")>]
 type CharacterSpan = 
 
-    val private _start: SnapshotPoint
+    val private _start: VirtualSnapshotPoint

```

Comment:
Did you consider breaking this into multiple types: CharacterSpan and VirtualCharacterSpan? That would give it symmetry with similar types. Possible that’s overkill here.

Reply:
There are many consumers of `CharacterSpan` that don't care if it's virtual or non-virtual; they just want to know the non-virtual interpretation of the span. This is analogous to caret position which is always encoded as a virtual point even when virtual space is not in use.

Likewise every non-virtual character span can be encoded as a virtual one with no loss of fidelity. The only disadvantage of using a single type is the same disadvantage as encoding the caret position as a virtual point: it is technically possible to encapsulate a span that is "more virtual" for the current `virtualedit` setting.